Modified challenge list

sanketbansal opened the pull request:
<!--
(Thanks for sending a pull request! .)
-->

<!--The PR title should start with "Fix #bugnum: " (if applicable), followed by a clear one-line present-tense summary of the changes introduced in the PR. For example: "Fix #bugnum: Introduce the first version of the collection editor." -->

<!-- Add issue number here. If you do not solve the issue entirely, please change the message e.g. "Addresses #IssueNumber -->

#### Changes proposed in this pull request:

<!-- Changes: Add here what changes were made in this issue and if possible provide links. -->
- Challenge list html file modified to make its UI similar to the current EvalAI application.
- Modified logic to display hosted challenges UI.
- Challenege list styles are changed by using the styles ported from current EvalAI project.
- Publiclist component and challenge card component are also modified correspondingly.
- Some utility component like side-bar and dash-footer component have been added.

<!-- Demo Link: Add here the link where you changes can be seen. -->
- Link to live demo: http://pr-143-evalai.surge.sh <!-- Replace XXX with your PR no: -->

<!-- Screenshots for the change: Add here the screenshot of the fix. -->

![Screenshot from 2019-06-11 13-01-23](https://user-images.githubusercontent.com/17106489/59292429-0ec8fe80-8c4b-11e9-9918-46aec6a7f0c4.png)

![Screenshot from 2019-06-11 13-01-30](https://user-images.githubusercontent.com/17106489/59292462-24d6bf00-8c4b-11e9-8302-0684ef335009.png)

![Screenshot from 2019-06-12 14-47-47](https://user-images.githubusercontent.com/17106489/59377889-1c50b800-8d21-11e9-8267-bd99241fe02d.png)

sanketbansal committed:
Ported image assets

Ported the image assets from angularJs application into 'image-port' directory

sanketbansal committed:
Added styles from angularjs

Ported styles from angularjs Application inside style-port folder

sanketbansal committed:
Modified auth service

Changes made in auth services to integrate angularJS authentication features

sanketbansal committed:
Modified angular.json file

Added main.scss file of ported style in the angular.json file to compile it at the time of building project.

sanketbansal committed:
Removed routing from auth service

Routing to 'auth/login' has been removed

sanketbansal committed:
Added side bar component

Added side bar as utility component for the side bar in the Dashboard UI

sanketbansal committed:
Added dash footer component

Dash footer added for improving dashboard UI

sanketbansal committed:
changes in style files

Added font variables in ported variable file
Added font style in ported main file.

sanketbansal committed:
Modified footer

Made the footer UI similar to the angularjs application one.

sanketbansal committed:
Added simple header component

simple header added to modify the UI of the header in the dashboard

sanketbansal committed:
Modified  Publiclist-page

Modified publiclist pages to make the UI  similar to current evalAiI

sanketbansal committed:
Modified Challenge card

Changes made in challenge card UI to make it similar to amgularjs UI.

sanketbansal committed:
Modified challenge list page.

Changes made in challenge list page UI using ported style and html.

sanketbansal committed:
Modifications in simple header component

Added color property for signup button
Added styles for login button

sanketbansal committed:
Modification in publiclist component

Added overflow property to fix bug in the UI
Removed the authentication logic from the component.

sanketbansal committed:
Modification in challenge list page

Added html for the hosted challenges
Added logic for authentication

sanketbansal committed:
Modifications in challenge card

Added the logic for hosted challenges UI.

sanketbansal commented:
@lunayach @Shekharrajak please review the PR.

lunayach commented:
Hi @RishabhJain2018, did we get the Surge to work properly? At this stage, both @sanketbansal and @Sanji515 are pushing PR's that are having UI changes (reproducing from EvalAI). They do attach some screenshots with a PR, but sometimes that may not be enough to see the working of the code entirely.

lunayach commented on src/app/components/nav/simple-header/simple-header.component.ts:
```diff
@@ -0,0 +1,150 @@
+  /**
+   * Inner width
+   */
+  public innerWidth: any;
```
(**minor**) This could be typed as a number, right? (Similar type changes could be done at some other places.)

lunayach commented on src/app/components/publiclists/publiclists.component.spec.ts:
```diff
@@ -28,6 +33,15 @@ const routes: Routes = [
       {path: 'me', component: ChallengelistComponent}
     ]
   },
+  {
```
@sanketbansal, If I'm correct, these changes are from an earlier PR, right?

Sanji515 replied:
yeah I think he worked on top of all his previous PR's and also pushed the previous code as well in all the PR's, that's why in all the PR's it about 35K lines of code added

Shekharrajak commented:
It is bad that you messed all the commits in all the PRs. But still you can improve your PRs.

1. One PR should be opened for one particular feature or fix.
2. If there are lot of things to do in one task, divide it into pieces and open PR
3. Making small PR will get more chances of early review, comments and merge.

Use `git` features to separate all irrelevant commits in all your PRs.

sanketbansal committed:
Moved styles to challenge-list component

Styles related to challenge-list component moved from 'styles-port/modules/challenge.scss'

sanketbansal committed:
Fixed lint error in angular.json file

Added comma after a property in angular.json file

sanketbansal committed:
Removed duplicate styles

Removed duplicate scss files from basic.scss

sanketbansal committed:
Removed styes and assets

Styles removed from styles-port folder
Assets removed from Images-port folder

sanketbansal committed:
Removed console logs

Unnecessary console logs removed

sanketbansal committed:
Added styles in styles-port folder

Added base and main styles.

sanketbansal committed:
Removed unused css

Removed unused css from main file

sanketbansal committed:
Removed comments

Removed comments from html files

sanketbansal committed:
Added component styles

Added component specific styles from main.scss file

sanketbansal committed:
changes in project configuration

Added basic.scss file globally.
Removed duplicate from app module file

sanketbansal committed:
Changes in styles-port folder

Removed main.scss file
Imported variable styles file basic.scss file

Shekharrajak commented:
I want to help you to improve your collaboration and Open Source Software development workflow skills but it looks like you don't want to follow the comments https://github.com/Cloud-CV/EvalAI-ngx/pull/143#issuecomment-503592390, https://github.com/Cloud-CV/EvalAI-ngx/pull/149#issuecomment-513453402

sanketbansal commented:
> I want to help you to improve your collaboration and Open Source Software development workflow skills but it looks like you don't want to follow the comments [#143 (comment)](https://github.com/Cloud-CV/EvalAI-ngx/pull/143#issuecomment-503592390), [#149 (comment)](https://github.com/Cloud-CV/EvalAI-ngx/pull/149#issuecomment-513453402)

I am working hard to improve these skills. The file changes in this PR will get reduced after merging #141 
As it has the dependencies on some changes present in that PR.

Shekharrajak commented:
Thanks for letting me know.

sanketbansal committed:
Removed styles-port folder

Style from basic.scss of styles-port folder has been removed

sanketbansal committed:
Removed simple header and dash footer

Removed unused simple header and dash footer

sanketbansal committed:
Changes in publiclist component

Added dash footer functionality using app-footer
Added simple header functionality using app-simple-header.

sanketbansal committed:
Changes in challenge list

Added style fw-light for font weight 300

sanketbansal committed:
Changes in footer component

Fixed bug in html file of footer component

lunayach commented on src/app/components/publiclists/challengelist/challengecard/challengecard.component.html:
```diff
@@ -1,55 +1,29 @@
+      <br> <strong>{{challenge['start_date'] | date:'medium'}} {{challenge['start_zone']}}</strong></p>
+    <p><strong class="text-light-black fs-12">Ends on</strong>
+      <br> <strong>{{challenge['end_date'] | date:'medium'}} {{challenge['end_zone']}}</strong></p>
+    <div class="margin-top-btm" *ngIf="this.routerPublic.url === '/challenges/me'">
```
Can you take `/challenges/me` from a constant variable in `.ts` file?

sanketbansal replied:
done

lunayach commented on src/app/components/publiclists/challengelist/challengelist.component.scss:
```diff
@@ -1,34 +1,190 @@
+}
+
+.bg-img {
+  /* height: 120%; */
```
Remove commented code. (Also, at some other places in this file.)

sanketbansal replied:
done

lunayach commented on src/app/components/publiclists/challengelist/challengelist.component.scss:
```diff
@@ -1,34 +1,190 @@
+  width: 100%;
+  margin: auto;
+  /* width: 109%; */
+  margin-top: -5%;
```
Define margin like this instead `  margin: -5% auto auto;` (remove the above one.)

sanketbansal replied:
done

lunayach commented on src/app/components/publiclists/challengelist/challengelist.component.scss:
```diff
@@ -1,34 +1,190 @@
+  background: $dark-gray;
+  width: 100%;
+  overflow: auto;
+  padding: 10px;
```
Remove this. You have already defined this below.

sanketbansal replied:
done

lunayach commented on src/app/components/publiclists/challengelist/challengelist.component.scss:
```diff
@@ -1,34 +1,190 @@
+.update-page {
+  display: block;
+  background: $dark-gray;
+  width: 100%;
```
Same, Remove this. Defined below.

sanketbansal replied:
done

lunayach commented on src/app/components/publiclists/challengelist/challengelist.component.ts:
```diff
@@ -119,15 +132,38 @@ export class ChallengelistComponent implements OnInit {
     }
+
+    this.authServiceSubscription = this.authService.change.subscribe((authState) => {
+      this.isLoggedIn = authState.isLoggedIn;
```
Define `'/auth/login'`, `/challenges/me'` as constants above. Check other such instances in this file/PR.

sanketbansal replied:
done

lunayach commented on src/app/components/publiclists/challengelist/challengelist.component.ts:
```diff
@@ -119,15 +132,38 @@ export class ChallengelistComponent implements OnInit {
+  @HostListener('window:scroll', [])
+  onWindowScroll(): void {
+    const RECT =  this.document.getElementById('ongoing-challenges').getBoundingClientRect();
+    if (RECT.top < 0) {
```
Simplify it like this, `this.isScrollbtnVisible = RECT.top < 0;`

sanketbansal replied:
done

lunayach commented on src/app/components/publiclists/publiclists.component.ts:
```diff
@@ -11,44 +12,22 @@ import { GlobalService } from '../../services/global.service';
-    this.scrollNav();
-    this.globalService.scrollToTop();
-  }
+  ngOnInit() {}
```
Remove this and the one below.

sanketbansal replied:
done

sanketbansal committed:
Changes in configuration

Removed unnecessary changes made in angular.json and app module files.

sanketbansal committed:
Changes in challenge list component

Added const route for '/challenge/me/'
Added shotrhand for margins
Removed unused scss from challenge list page
Added common route paths for routing .
Fixed bugs in the UI

lunayach commented on src/app/components/publiclists/challengelist/challengelist.component.html:
```diff
@@ -1,14 +1,63 @@
+  </div>
+
+  <!-- past challenges -->
+  <div *ngIf="routerPublic.url === '/challenges/all'">
```
Get `url` from a constant in `.ts` file. Also, just cross-check this thing for the whole file.

sanketbansal replied:
done

lunayach reviewed (approved):
@RishabhJain2018 This looks good to be merged!

sanketbansal committed:
Merge branch 'master' into modified-challengeList
